build_all/mac/release.sh

```diff
@@ -45,96 +51,138 @@ process_args()
-export OPENSSL_ROOT_DIR=/usr/local/opt/openssl
-c_build_root=${build_root}"/c"
-
-# brew installs python 3.x to $prefix/include/python3.xm
-if [ $PYTHON_VERSION != "3.4" ] && [ $PYTHON_VERSION != "3.5" ] && [ $PYTHON_VERSION != "3.6" ] && [ $PYTHON_VERSION != "3.7" ]
-then
-    python_prefix=$(python-config --prefix)
-    python_include=$python_prefix/include/python$PYTHON_VERSION
-    python_lib=$python_prefix/lib/libpython$PYTHON_VERSION.dylib
-else
-    python_prefix=$(python3-config --prefix)
-    python_include=$python_prefix/include/python${PYTHON_VERSION}m
-    python_lib=$python_prefix/lib/libpython${PYTHON_VERSION}m.dylib
-fi
-
-
-rm -r -f $build_folder
-mkdir -p $build_folder
-pushd $build_folder
-cmake -Drun_valgrind:BOOL=OFF -DcompileOption_CXX:STRING="-Wno-unused-value" -DcompileOption_C:STRING="-Wno-unused-value" -Drun_e2e_tests:BOOL=OFF -Drun_longhaul_tests=OFF -Duse_amqp:BOOL=ON -Duse_http:BOOL=ON -Duse_mqtt:BOOL=ON -Ddont_use_uploadtoblob:BOOL=OFF -Duse_wsio:BOOL=ON -Drun_unittests:BOOL=OFF -Dbuild_python:STRING=$PYTHON_VERSION -Dbuild_javawrapper:BOOL=OFF -Dno_logging:BOOL=OFF $c_build_root -Dwip_use_c2d_amqp_methods:BOOL=OFF -DPYTHON_LIBRARY=$python_lib -DPYTHON_INCLUDE_DIR=$python_include -Duse_prov_client=OFF -Duse_edge_modules=ON -Dskip_samples=ON
-
-# Set the default cores
-CORES=$(sysctl -n hw.ncpu)
+build_c_sdk()
+{
+    # build azure-iot-sdk-c
+    export OPENSSL_ROOT_DIR=/usr/local/opt/openssl
+    c_build_root=${build_root}"/c"
+
+    # brew installs python 3.x to $prefix/include/python3.xm
+    if [ $PYTHON_VERSION != "3.4" ] && [ $PYTHON_VERSION != "3.5" ] && [ $PYTHON_VERSION != "3.6" ] && [ $PYTHON_VERSION != "3.7" ]
+    then
+        python_prefix=$(python-config --prefix)
+        python_include=$python_prefix/include/python$PYTHON_VERSION
+        python_lib=$python_prefix/lib/libpython$PYTHON_VERSION.dylib
+    else
+        python_prefix=$(python3-config --prefix)
+        python_include=$python_prefix/include/python${PYTHON_VERSION}m
+        python_lib=$python_prefix/lib/libpython${PYTHON_VERSION}m.dylib
+    fi
+
+
+    rm -r -f $build_folder
+    mkdir -p $build_folder
+    pushd $build_folder
+    cmake -Drun_valgrind:BOOL=OFF -DcompileOption_CXX:STRING="-Wno-unused-value" -DcompileOption_C:STRING="-Wno-unused-value" -Drun_e2e_tests:BOOL=OFF -Drun_longhaul_tests=OFF -Duse_amqp:BOOL=ON -Duse_http:BOOL=ON -Duse_mqtt:BOOL=ON -Ddont_use_uploadtoblob:BOOL=OFF -Duse_wsio:BOOL=ON -Drun_unittests:BOOL=OFF -Dbuild_python:STRING=$PYTHON_VERSION -Dbuild_javawrapper:BOOL=OFF -Dno_logging:BOOL=OFF $c_build_root -Dwip_use_c2d_amqp_methods:BOOL=OFF -DPYTHON_LIBRARY=$python_lib -DPYTHON_INCLUDE_DIR=$python_include -Duse_prov_client=OFF -Duse_edge_modules=ON -Dskip_samples=ON
+
+    # Set the default cores
+    CORES=$(sysctl -n hw.ncpu)
+
+    make --jobs=$CORES
+    ctest -j $CORES -C "Debug" --output-on-failure
+
+    popd
+}
 
-make --jobs=$CORES
-ctest -j $CORES -C "Debug" --output-on-failure
+copy_binaries()
```

Comment:
needs error checking

Reply:
When we manually run the release script we always have to check the output for errors.

Reply:
just to clarify i'm asking for more of this: `[ $? -eq 0 ] || exit $?`
it's already used in some parts of the scripts, but not all of them - why the inconsistency?
should pretty much be everywhere, right?

having a human check console output is error prone.

Reply:
Added error handling